compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -955,6 +960,92 @@ impl<'a> Parser<'a> {
         }
     }
 
+    /// Recover from `Fn`-family traits (Fn, FnMut, FnOnce) with lifetime arguments
+    /// (e.g. `FnOnce<'a>(&'a str) -> bool`). Up to generic arguments have already
+    /// been eaten.
+    fn recover_fn_trait_with_lifetime_params(
+        &mut self,
+        fn_path: &mut ast::Path,
+        lifetime_defs: &mut Vec<GenericParam>,
+    ) -> PResult<'a, ()> {
+        let fn_path_segment = fn_path.segments.last_mut().unwrap();
+        let generic_args = if let Some(p_args) = &fn_path_segment.args {
+            p_args.clone().into_inner()
+        } else {
```

Comment:
This can now be

```suggestion
        let Some(p_args) = &fn_path_segment.args {
            p_args.clone().into_inner()
        } else {
```

Reply:
Is this valid syntax? I got the following error:
```
error: expected one of `(`, `.`, `::`, `;`, `?`, `else`, or an operator, found `{`
    --> compiler/rustc_parse/src/parser/ty.rs:1033:50
     |
1033 |         let Some(p_args) = &fn_path_segment.args {
     |                                                  ^ expected one of 7 possible tokens

error: could not compile `rustc_parse` due to previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `rustc_parse` due to previous error
```

Reply:
With `let`/`else` syntax, it's

```rust
let Some(p_args) = &fn_path_segment.args else {
    // ... ... ...
    return Ok(());
};
let p_args = p_args.clone().into_inner();
```

or

```rust
let Some(p_args) = fn_path_segment.args.cloned().map(|arg| arg) else {
    // ... ... ...
    return Ok(());
};
```

or

```rust
let Some(p_args) = fn_path.segments
    .last_mut()
    .unwrap()
    .args
    .cloned()
    .map(|arg| arg)
else {
    // ... ... ...
    return Ok(());
};
```

at your choosing.

Reply:
Ah, yeah, I mistyped what I meant. Thanks @fmease for clarifying it